lib/ayesql/compiler.ex

```diff
@@ -181,7 +189,47 @@ defmodule AyeSQL.Compiler do
   end
 
   @spec create_query(query()) :: Macro.t()
-  defp create_query({name, docs, fragments}) do
+  defp create_query({name, :script, docs, fragments}) do
+    fragments = Macro.escape(fragments)
+
+    quote do
+      @doc AyeSQL.Compiler.gen_docs(unquote(docs), unquote(fragments))
+      @spec unquote(name)(AyeSQL.Core.parameters()) ::
+              {:ok, AyeSQL.Query.t() | term()}
+              | {:error, AyeSQL.Error.t() | term()}
+      @spec unquote(name)(AyeSQL.Core.parameters(), AyeSQL.Core.options()) ::
+              {:ok, AyeSQL.Query.t() | term()}
+              | {:error, AyeSQL.Error.t() | term()}
+      def unquote(name)(params, options \\ [])
+
+      def unquote(name)(params, options) do
+        options = Keyword.merge(__MODULE__.__db_options__(), options)
+
+        {index, options} = Keyword.pop(options, :index, 1)
+        {run?, options} = Keyword.pop(options, :run, true)
+
+        content = AyeSQL.AST.expand(__MODULE__, unquote(fragments))
+        context = AyeSQL.AST.Context.new(index: index)
+
+        with {:ok, %{statement: statement} = query} <-
+               AyeSQL.Core.evaluate(content, params, context) do
+          if run? do
+            statement
+            |> String.split(";", trim: true)
+            |> Enum.map(&String.trim(&1))
+            |> Enum.each(&__MODULE__.run!(%{query | statement: &1}, options))
```

Comment:
@dkuku Splitting a statement with `;` and then running each statement by itself makes the script operation unsafe as it doesn't happen inside a transaction.

Instead of this change, wouldn't it be better to use an actual script and use Postgrex or Ecto's APIs to do the sustitution?